plumbing/transport/file/server.go

```diff
@@ -0,0 +1,82 @@
+package file
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/server"
+	"gopkg.in/src-d/go-git.v4/storage/filesystem"
+	"gopkg.in/src-d/go-git.v4/utils/ioutil"
+
+	osfs "srcd.works/go-billy.v1/os"
+)
+
+var DefaultServer = NewServer(server.DefaultHandler)
+
+type Server struct {
```

Comment:
You can call Serve multiple times with different arguments for different repositories.
I am thinking of modularizing it further by decoupling repository finding/loading from the server itself, just as with the server.Handler. Something like server.StorerLoader. The default one would load repositories by file system path and could be shared by default implementations, but the user could pass a loader getting repositories from a database. What do you think?